openwisp_monitoring/device/apps.py

```diff
@@ -11,3 +15,57 @@ class DeviceMonitoringConfig(AppConfig):
 
     def ready(self):
         manage_short_retention_policy()
+        from openwisp_controller.connection.models import DeviceConnection
+        from openwisp_controller.connection.signals import is_working_changed
+        is_working_changed.connect(self.health_status_changed,
+                                   sender=DeviceConnection,
+                                   dispatch_uid='change_health_status')
+
+    @classmethod
+    def health_status_changed(cls, instance, is_working, **kwargs):
+        from ..check.models import Check
+        d = instance.device
+        dm = d.monitoring
```

Comment:
More descriptive identifiers will make it more readable.

Reply:
I am not sure about this, it appears all right to me

Reply:
do not use abbreviated variable names in non test code please